### ISSUE TITLE
torch.compile fails on second-order gradient after autograd.grad(create_graph=True)

### ISSUE THREAD

**Issue by ALinrunrun** [pytorch/pytorch#183603]
### 🐛 Describe the bug

`torch.compile` with the Inductor backend fails when computing a second-order gradient from a function that internally calls `torch.autograd.grad(..., create_graph=True)`.

Eager execution successfully computes the second derivative, but the compiled version raises a `RuntimeError` when calling `torch.autograd.grad` on the compiled output.

```
import torch

def f(x):
    y = (x ** 4).sum()
    return torch.autograd.grad(y, x, create_graph=True)[0]

torch._dynamo.reset()

xe = torch.tensor([0.5, 1.5, 2.5, 3.5], requires_grad=True)
ge = f(xe)
he = torch.autograd.grad(ge.sum(), xe)[0]
print("eager d2:", he.tolist())

torch._dynamo.reset()

xc = torch.tensor([0.5, 1.5, 2.5, 3.5], requires_grad=True)
gc = torch.compile(f, backend="inductor")(xc)

try:
    hc = torch.autograd.grad(gc.sum(), xc)[0]
    print("compile d2:", hc.tolist())
except Exception as e:
    print("compile error:", type(e).__name__)
```

### Error logs

eager d2: [3.0, 27.0, 75.0, 147.0]
compile error: RuntimeError

### Versions

PyTorch version: 2.11.0+cu130
Is debug build: False
CUDA used to build PyTorch: 13.0
ROCM used to build PyTorch: N/A

OS: Ubuntu 24.04.2 LTS (x86_64)
GCC version: (Ubuntu 13.3.0-6ubuntu2~24.04.1) 13.3.0
Clang version: Could not collect
CMake version: Could not collect
Libc version: glibc-2.39

Python version: 3.10.20 (main, Mar 11 2026, 17:46:40) [GCC 14.3.0] (64-bit runtime)
Python platform: Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.39
Is CUDA available: False
CUDA runtime version: Could not collect
CUDA_MODULE_LOADING set to: N/A
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3080 Laptop GPU
Nvidia driver version: 545.92
cuDNN version: Could not collect
Is XPU available: False
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True
Caching allocator config: N/A

Versions of relevant libraries:
[pip3] numpy==2.2.6
[pip3] onnx==1.21.0
[pip3] onnx2torch==1.5.15
[pip3] onnxruntime==1.23.2
[pip3] torch==2.11.0
[pip3] torchvision==0.26.0
[pip3] triton==3.6.0

cc @chauhang @penguinwu

**Comment by khushi-411** [comment]
Hi, I think this is a duplicate of #181581

**Comment by jansel** [comment]
My agent says this is now intended behavior on current main. The duplicate issue #181581 was fixed by #181606 by preserving the create_graph=True gradient metadata, but actual AOTAutograd double backward remains explicitly unsupported and now raises `torch.compile with aot_autograd does not currently support double backward` at the second backward. The local repro no longer silently drops grad metadata; it reaches that intended unsupported-feature error instead.
